core/common/src/main/java/alluxio/metrics/Metric.java

```diff
@@ -275,4 +296,35 @@ public String toString() {
         .add("hostname", mHostname).add("instanceId", mInstanceId).add("name", mName)
         .add("value", mValue).add("tags", mTags).toString();
   }
+
+  /**
+   * Data structure representing a metric name and user name.
+   */
+  private static class UserKey {
```

Comment:
perhaps call this `MetricKey`? It's a key to determine the name for a metric

Reply:
`UserMetricKey`?

Reply:
deal